### PR TITLE
Red 1953 pagination

### PR DESCRIPTION
**Pull request by ecoologic**
Implement the solution that works with all types of pagination (obp / cbp / single page). Defaults to CBP.

Screenshot 2023-10-27 at 9 35 05 am

** Update: I extracted the iterator in its own trait.

### CODE REVIEW

File: src/Zendesk/API/Resources/Core/Tickets.php [hunk 1]
```diff
@@ -45,19 +44,6 @@ class Tickets extends ResourceAbstract
...
-     *     process($ticket)
-     * }
-     *
-     * @return TicketsIterator Returns a new TicketsIterator object.
-     */
-    public function iterator()
-    {
-        return new TicketsIterator($this);
-    }
-
```
Comment (ecoologic) [inline src/Zendesk/API/Resources/Core/Tickets.php:60]: moved into trait to make it generic to all resources.

File: src/Zendesk/API/Resources/Core/AppInstallations.php [hunk 2]
```diff
@@ -123,4 +124,8 @@ public function update($id = null, array $updateResourceFields = [], $routeKey =
             $updateResourceFields
         );
     }
+
+    private function paginationStrategyClass() {
+        return ObpStrategy::class;
+    }
```
Comment (ecoologic) [inline src/Zendesk/API/Resources/Core/AppInstallations.php:130]: const are not available in the current PHP version (7.4)

File: src/Zendesk/API/Traits/Resource/FindAll.php [hunk 3]
```diff
@@ -35,4 +37,30 @@ public function findAll(array $params = [], $routeKey = __FUNCTION__)
...
+    {
+        $strategyClass = $this->paginationStrategyClass();
+        $strategy = new $strategyClass($this, $this->resourcesRoot(), 2);
+        return new PaginationIterator($strategy);
+    }
+
+
+    private function paginationStrategyClass() {
+        return CbpStrategy::class;
+    }
```
Comment (ecoologic) [inline src/Zendesk/API/Traits/Resource/FindAll.php:61]: This will act as default

File: docker-compose.yml [hunk 4]
```diff
@@ -8,6 +8,9 @@ services:
       - .:/app
       - vendor:/app/vendor
     command: vendor/bin/phpunit --testsuite "Zendesk API Unit Test Suites"
+    # command: vendor/bin/phpunit tests/Zendesk/API/UnitTests/Traits/Utility/PaginationTest.php
+    # command: vendor/bin/phpunit tests/Zendesk/API/UnitTests/Core/TicketsTest.php
+    # command: vendor/bin/phpunit tests/Zendesk/API/UnitTests/Core/OrganizationMembershipsTest.php
```
Comment (ecoologic) [inline docker-compose.yml:13]: I'll do some cleaning at the end, but these comments come useful to switch between tests.

**Review by fbvilela — APPROVED** [review]

**Pull request merged**